box-share-sdk/src/main/java/com/box/androidsdk/share/api/BoxShareController.java

```diff
@@ -0,0 +1,160 @@
+package com.box.androidsdk.share.api;
+
+import com.box.androidsdk.content.BoxApiBookmark;
+import com.box.androidsdk.content.BoxApiCollaboration;
+import com.box.androidsdk.content.BoxApiFile;
+import com.box.androidsdk.content.BoxApiFolder;
+import com.box.androidsdk.content.BoxFutureTask;
+import com.box.androidsdk.content.models.BoxBookmark;
+import com.box.androidsdk.content.models.BoxCollaboration;
+import com.box.androidsdk.content.models.BoxFile;
+import com.box.androidsdk.content.models.BoxFolder;
+import com.box.androidsdk.content.models.BoxItem;
+import com.box.androidsdk.content.models.BoxIteratorCollaborations;
+import com.box.androidsdk.content.models.BoxObject;
+import com.box.androidsdk.content.models.BoxSharedLink;
+import com.box.androidsdk.content.models.BoxVoid;
+import com.box.androidsdk.content.requests.BoxRequest;
+import com.box.androidsdk.content.requests.BoxRequestBatch;
+import com.box.androidsdk.content.requests.BoxRequestUpdateSharedItem;
+import com.box.androidsdk.content.requests.BoxRequestsFile;
+import com.box.androidsdk.content.requests.BoxResponseBatch;
+import com.box.androidsdk.content.utils.SdkUtils;
+import com.box.androidsdk.internal.BoxApiInvitee;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by varungupta on 3/4/2016.
+ */
+public class BoxShareController implements ShareController {
+    private BoxApiFile mFileApi;
+    private BoxApiFolder mFolderApi;
+    private BoxApiBookmark mBookmarkApi;
+    private BoxApiCollaboration mCollabApi;
+
+    public BoxShareController(BoxApiFile fileApi, BoxApiFolder folderApi, BoxApiBookmark bookmarkApi, BoxApiCollaboration collaborationApi) {
+        mFileApi = fileApi;
+        mFolderApi = folderApi;
+        mBookmarkApi = bookmarkApi;
+        mCollabApi = collaborationApi;
+    }
+
+    @Override
+    public void fetchItemInfo(BoxItem boxItem, BoxFutureTask.OnCompletedListener<BoxItem> onCompletedListener) {
```

Comment:
I think it is cleaner here if you just return the task for these instead of requiring a completed listener to be passed in. The content sdk already supports adding the listener if needed, and this gives the added flexibility of calling .get to block on the task.